### DISCUSSION TITLE
Support for OBJ+MTL from Fusion360?

### DISCUSSION THREAD

**Discussion by helicopter12** [bulletphysics/bullet3#4055] [Q&A] [answered]
Hello, 

I am using Python 3.9.9 as well as PyBullet 3.1.7 / Bullet Physics 3.17 and playing around with an Oculus Rift VR headset. I am exporting models that I have created from Fusion 360 as .Obj files (which then come with a .mtl materials file for colors, etc) but when I import these files into Pybullet using loadURDF, the colors from the .mtl file are not used. I know the .obj file and .mtl file are working because I can open the .obj file with Windows "3D Viewer" program and it shows the object colored correctly. 

The URDF file I am using looks as follows:
```
<?xml version="1.0" ?>
<robot name="cube">
  <link name="light.urdf">
    <contact>
      <lateral_friction value=".5"/>
    </contact>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
       <mass value=".05"/>
       <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
				<mesh filename="light.obj" scale="1 1 1"/>
      </geometry>
	  <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
	 	<box size="0.15 0.05 0.03"/>
      </geometry>
    </collision>
  </link>
</robot>
```
Here are the .obj files and .mtl files for reference:
[light.zip](https://github.com/bulletphysics/bullet3/files/7629964/light.zip)

This is what I expect:
![image](https://user-images.githubusercontent.com/2042120/144155558-e04ce967-80e4-4dad-bcd1-7385f5bd26da.png)

This is what I get:
![image](https://user-images.githubusercontent.com/2042120/144155625-1dfcbe5b-c7aa-46e9-a90b-33ed59f3007a.png)

Has anyone run into similar issues or is this not a feature and I am just mistaken?

**Comment by rohit-kumar-j** [comment, marked as answer]
@helicopter12 See #3581  (Perhaps this can solve the issue?)

>PyBullet (and the underlying Bullet Robotics APIs) rendering has some limitations:
>
>  1. wavefront obj files can have at most 1 texture
  >2. there is a limited pre-allocated memory for visual meshes

**Reply by helicopter12** [reply]
@rohit-kumar-j 
Very helpful, thanks so much; I will play around with these ideas and see if I can make my models behave as I want them too!
